### DISCUSSION TITLE
Is it reasonable that i take the harmony embedding as the input of the assemble function in low dimensional space for seeking a corrected matrix?

### DISCUSSION THREAD

**Discussion by WeissTsang** [brianhie/scanorama#125] [Q&A]
Hi @brianhie,

I was wondering whether it is reasonable that i take the harmony (which is another batch correcting method) embedding and the preprocessed data as the inputs of the assemble function for seeking a corrected matrix. Cuz scanorama hardly removes the batch effect of my datasets, while the harmony performs better. However, harmony does not calculate a corrected expression matrix for genes. So is it possible that i combine assemble function in scanorama with harmony embedding to get a corrected matrix?

Thank you in advance for any advice or recommendation.
Best regards,
Weiss

**Comment by brianhie** [comment]
Hi @WeissTsang, Scanorama is pretty conservative in its integration, which looks to be the case for your data. If the Harmony integration makes sense (especially if you have positive/negative controls indicating integration is working as expected), it would be fine to just use those reduced dimensions.

Regarding the values in the full corrected gene matrix, I would not interpret those as true gene expression values (the informative information is in the distances between cells, not the values themselves). You could try a conservative batch correction tool like Combat if you want to do downstream analysis in gene expression space.
